eval/eval-scripts/mmlu.py

```diff
@@ -0,0 +1,138 @@
+    model_s3_path, output_file, output_s3_path, model_dtype="bfloat16", few_shots=5, batch_size=8, device=None
+):
+    # These env vars must be set in the pipeline.yaml
+    aws_access_key = os.getenv("AWS_ACCESS_KEY_ID", "")
+    aws_secret_key = os.getenv("AWS_SECRET_ACCESS_KEY", "")
+    aws_region = os.getenv("AWS_REGION", "")
+    local_model_path = os.getenv("LOCAL_MODEL_PATH", "/tmp/rc")
+    bucket_name = os.getenv('BUCKET_NAME')
+    bucket_host = os.getenv('BUCKET_HOST')
+
+    # debug
+    logger.debug(f"LOCAL_MODEL_PATH: {local_model_path}")
+
+    # MMLU_TASKS defaults to "all tasks" when not set
+    mmlu_tasks_env = os.getenv("MMLU_TASKS", "")
+    if mmlu_tasks_env:
+        mmlu_tasks = mmlu_tasks_env.split(',')
+    else:
+        mmlu_tasks = MMLU_TASKS 
+
+    # TODO: Update this
+    s3_endpoint_url = f'http://{bucket_host}'
+
+    os.environ["PYTORCH_CUDA_ALLOC_CONF"] = "expandable_segments:True"
+
+    s3_client = boto3.client(
+        "s3",
+        aws_access_key_id=aws_access_key,
+        aws_secret_access_key=aws_secret_key,
+        region_name=aws_region,
+        endpoint_url=s3_endpoint_url,
+    )
+
+    timestamp = datetime.now().strftime("%Y%m%d_%H%M%S")
+    output_file = f"{os.path.splitext(output_file)[0]}_{timestamp}{os.path.splitext(output_file)[1]}"
+
+    required_files = ['config.json']
+    existing_files = os.listdir(local_model_path) if os.path.exists(local_model_path) else []
+
+    # Check if all required files are present
+    if all(file in existing_files for file in required_files):
+        logger.debug(f"Model directory {local_model_path} contains required files. Skipping download.")
+    else:
+        os.makedirs(local_model_path, exist_ok=True)
+        logger.debug(f"Downloading model to {local_model_path} from {model_s3_path}")
+
+        try:
+            response = s3_client.list_objects_v2(Bucket=bucket_name, Prefix=model_s3_path)
+            if 'Contents' not in response:
+                logger.debug(f"No objects found at {model_s3_path}. Check the S3 path and permissions.")
+                return
+
+            for obj in response['Contents']:
+                key = obj['Key']
+                local_file_path = os.path.join(local_model_path, os.path.basename(key))
+                s3_client.download_file(bucket_name, key, local_file_path)
+                logger.debug(f"Downloaded {key} to {local_file_path}")
+
+            # Verify that the expected files are downloaded
+            existing_files = os.listdir(local_model_path)
+            if all(file in existing_files for file in required_files):
+                logger.debug(f"Required model files found at {local_model_path}.")
+            else:
+                logger.debug(f"Model files not found at {local_model_path}. Expected files: {required_files}. Check the S3 path and permissions.")
+
+        except Exception as e:
+            logger.debug(f"Error downloading model from S3: {e}")
+            raise
```

Comment:
Can't you use here `dsl.importer` instead? It will be safer + less code to maintain + always using tracked artifacts.

https://www.kubeflow.org/docs/components/pipelines/user-guides/components/importer-component/

Reply:
yup

> WIP: Don't Review!! Don't try to run (probably won't run anyways) :)